build_tools/integration_tests/llm/cpu_llm_server_test.py

```diff
@@ -82,4 +82,8 @@ def test_llm_server(llm_server, available_port):
     assert llm_server.poll() is None
     output = do_generate("1 2 3 4 5 ", available_port)
     logger.info(output)
-    assert output.startswith("6 7 8")
+    # TODO: Remove when fixed: https://github.com/nod-ai/SHARK-Platform/issues/437
+    with pytest.xfail(
+        reason="Accuracy with Shortfin LLM Server is decreased from latest iree-compiler RC."
+    ):
+        assert output.startswith("6 7 8")
```

Comment:
This is fine, but here are some other ideas to consider:

```python
# add this to the top of the test file, a conftest.py, or a utils file
class AccuracyValidationException(RuntimeError):
    pass

# put this in the test body
expected_output_prefix = "6 7 8"
if not output.startswith(expected_output_prefix):
  raise AccuracyValidationException(f"Expected '{output}' to start with '{expected_output_prefix}'")

# Put this on the `def test_llm_server`
@pytest.mark.xfail(raises=AccuracyValidationException)
```

I think that would give a bit more context in the logs when the test starts passing again, or fails for a different reason

Reply:
Updated